log/src/main/java/keywhiz/log/Event.java

```diff
@@ -0,0 +1,68 @@
+package keywhiz.log;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A class to represent an event in Keywhiz
+ */
+public class Event {
+  private final Instant timestamp;
+  private final EventTag type;
+  private final String user;
+  private final String objectName; // The name of the affected object
+  private final Map<String, String> extraInfo; // Any extra information
+
+  private final DateTimeFormatter df =
+      DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss").withZone(ZoneId.of("UTC"));
+
+  public Event(Instant timestamp, EventTag type, String user, String objectName,
+      @Nullable Map<String, String> extraInfo) {
```

Comment:
You can have two constructors, one with the `extraInfo` argument and one without.